cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/model/ClusterModelStats.java

```diff
@@ -218,28 +225,9 @@ public String getJSONString() {
    * @return An object that can be further used to encode into JSON.
    */
   public Map<String, Object> getJsonStructure() {
-    Map<String, Object> statMap = new HashMap<>();
-    Map<String, Integer> basicMap = new HashMap<>();
-    basicMap.put(AnalyzerUtils.BROKERS, numBrokers());
-    basicMap.put(AnalyzerUtils.REPLICAS, numReplicasInCluster());
-    basicMap.put(AnalyzerUtils.TOPICS, numTopics());
-    List<Statistic> cachedStatistic = Statistic.cachedValues();
-    // List of all statistics AVG, MAX, MIN, STD
-    Map<String, Object> allStatMap = new HashMap<>(cachedStatistic.size());
-    for (Statistic stat : cachedStatistic) {
-      List<Resource> cachedResources = Resource.cachedValues();
-      Map<String, Object> resourceMap = new HashMap<>(cachedResources.size() + 3);
-      for (Resource resource : cachedResources) {
-        resourceMap.put(resource.resource(), resourceUtilizationStats().get(stat).get(resource));
-      }
-      resourceMap.put(AnalyzerUtils.POTENTIAL_NW_OUT, potentialNwOutUtilizationStats().get(stat));
-      resourceMap.put(AnalyzerUtils.REPLICAS, replicaStats().get(stat));
-      resourceMap.put(AnalyzerUtils.LEADER_REPLICAS, leaderReplicaStats().get(stat));
-      resourceMap.put(AnalyzerUtils.TOPIC_REPLICAS, topicReplicaStats().get(stat));
-      allStatMap.put(stat.stat(), resourceMap);
-    }
-    statMap.put(AnalyzerUtils.METADATA, basicMap);
-    statMap.put(AnalyzerUtils.STATISTICS, allStatMap);
+    Map<String, Object> statMap = new HashMap<>(2);
+    statMap.put(METADATA, new ClusterModelStatsMetaData().getJsonStructure());
```

Comment:
Let's avoid backward incompatible name changes.

Reply:
Here I actually refactor out the sub layer of JSON response into another Classes(`ClusterModelStatsMetaData` and `  ClusterModelStatsValue` here), the actual data layout does not change.

The reason I do this is because it is hard to annotate a class if it contains keys of multiple JSON layers, in that case we need to distinguish which layer the key should resides in.